### PR TITLE
Add Fetch monitoring to Dependencies Plugin

### PR DESCRIPTION
**Pull request by markwolff**

### CODE REVIEW

File: AISKU/src/Initialization.ts [hunk 1]
```diff
@@ -198,7 +213,7 @@ export class Initialization implements IApplicationInsights {
 
         extensions.push(appInsightsChannel);
         extensions.push(this.properties);
-        extensions.push(this.dependencies);
+        // extensions.push(this.dependencies);
```
Comment (jpiyali) [inline AISKU/src/Initialization.ts]: if you dont add to extensions, you wont called through .initialize()

File: AppInsightsCommon/src/Telemetry/Common/DataSanitizer.ts [hunk 2]
```diff
@@ -179,4 +178,12 @@ export class DataSanitizer {
         var s = "00" + num;
         return s.substr(s.length - 3);
     }
+
+    /**
+     * helper method to trim strings (IE8 does not implement String.prototype.trim)
+     */
+    public static trim(str: any): string {
```
Comment (jpiyali) [inline AppInsightsCommon/src/Telemetry/Common/DataSanitizer.ts]: datasanitizer in common has it, maybe later expose it.

File: extensions/applicationinsights-dependencies-js/src/ajax.ts [hunk 3]
```diff
@@ -17,20 +17,39 @@ export interface XMLHttpRequestInstrumented extends XMLHttpRequest {
...
+    trackDependencyData(dependency: IDependencyTelemetry, properties?: { [key: string]: any });
 }
 
-export class AjaxMonitor implements ITelemetryPlugin, IDependenciesPlugin {
-    private initialized: boolean;
+export interface IInstrumentationRequirements extends IDependenciesPlugin {
+    includeCorrelationHeaders: (ajaxData: ajaxRecord, input?: Request | string, init?: RequestInit, xhr?: XMLHttpRequestInstrumented) => any;
+}
+
+export class AjaxMonitor implements ITelemetryPlugin, IDependenciesPlugin, IInstrumentationRequirements {
```
Comment (jpiyali) [inline extensions/applicationinsights-dependencies-js/src/ajax.ts]: Noneed to implement IDependenciesPlugin separately.

File: AppInsightsCommon/src/Telemetry/Common/DataSanitizer.ts (hunk 2 above)
Comment (markwolff) [reply AppInsightsCommon/src/Telemetry/Common/DataSanitizer.ts]: This is just a hack to remove common's circular dependency. Can look into a better solution in the future

**Pull request merged**